templates/overcloud-templates.yml.j2

```diff
@@ -2,11 +2,7 @@
 {% if extra_templates is defined and extra_templates %}
 tripleo_heat_templates:
 {% for template in  env_files.results %}
-{% if template.stat.exists == False %}
     - /home/stack/{{ template.item }}
-{% else %}
-    - {{ template.item }}
```

Comment:
@smalleni e.g if the template.stat.exists == true it not able to find the firstboot.yaml file but if you specify it to be /home/stack/{{ template.item }} it resolves the issue and this fixes #136

Reply:
1.what if the file doesn't exist there too?
2.what if the emplate.stat.exists == true is true?

i guess we need to swap the else part to true and provide the fully qualified name in the all.yaml.(ignore if it is wrong :-) )

Reply:
for the first time when we run an overcloud.yml the env_files.exists=false so we copy the firstboot.yaml to undercloud and then when we template out extra overcloud-templates [file](https://github.com/redhat-performance/jetpack/blob/master/templates/overcloud-templates.yml.j2) the path for tripleo_heat_templates is /home/stack//{{ template.item }}. But when we run it for the next time the firstboot.yml already exists on undercloud  as it was copied previously so the env_files.exists=true but due to the overcloud-templates.yml.j2 it tries to find the tripleo_heat_templates at {{ template.item }} . Also this path is passed in the overcloud_deploy.sh so the file should be located on undercloud i.e /home/stack/{{ template.item }}

Reply:
I will explain how this works and then you will understand why we need to take a different approach.

In this task at https://github.com/redhat-performance/jetpack/blob/master/overcloud.yml#L33 we check if the templates in the extra_templates in group_vars/all.yml exist on the undercloud.

The extra_templates variable list can have 
1. a relative path to an extra THT file which is in the jetpack directory on the host running jetpack (jump host) like firstboot.yaml that we currently have
2. an absolute path to a file like /usr/share/openstack-tripleo-heat-templates/environments/services/neutron-ovs-dpdk.yaml that exists on the undercloud node

Depending on whether or not the files in the extra_templates list exist on the undercloud, they are copied from the local directory of jetpack on the jumphost  to /home/stack/ on the undercloud if they do not exist. And then {{ infrared_dir }}/plugins/tripleo-overcloud/vars/overcloud/templates/extra.yml is templated out based on the location of file (absolute path if file exists or /home/stack if the file was copied from jetpack directory)

The issue described in https://github.com/redhat-performance/jetpack/issues/136 happens when
1. The THT file in the local jetpack directory on the ansible jump host has already been copied to /home/stack/<file.yaml> as it does not exist on the undercloud and then the playbook main.yml failed at some place. 
2. The user reruns the overcloud install by using ansible-playbook main.yaml --tags overcloud
3.  When the task at  https://github.com/redhat-performance/jetpack/blob/master/overcloud.yml#L35 runs, it runs as the stack user, so although it is searching for the firstboot.yaml relative path, it gets translated to /home/stack/firstboot.yml and finds the absolute file, so when it templates out the {{ infrared_dir }}/plugins/tripleo-overcloud/vars/overcloud/templates/extra.yml , it places "firstboot.yaml"  in the file, which is not the absolute path, so it does not resolve correctly. Bottom line is the extra.yml file we need absolute paths.

While you patch fixes the problem by always making path /home/stack/filename.yaml, it does not let user specify an existing THT file on the undercloud  in /usr/share/openstack-tripleo-heat-templates/.....

Does that make sense? We are losing the flexibility of specifying existing files on the undercloud. This patch to be re-worked and the problem might better be solved by changes in this template and/or tasks in the block at https://github.com/redhat-performance/jetpack/blob/master/overcloud.yml#L34

Reply:
Have updated the PR so that it fixes #136 and also provides the user the flexibility of specifying the existing files on undercloud